pkg/unlock.go

```diff
@@ -71,62 +60,70 @@ func NewCmdUnlockRepository(clientGetter genericclioptions.RESTClientGetter) *co
 			if err != nil {
 				return err
 			}
-			// unlock local backend
-			if repository.Spec.Backend.Local != nil {
-				if err = unlockLocalRepo(kc, repository); err != nil {
-					return fmt.Errorf("can't unlock repository for local backend, reason: %s", err)
-				}
-				return nil
-			}
+
 			// get source repository secret
 			secret, err := kc.CoreV1().Secrets(namespace).Get(repository.Spec.Backend.StorageSecretName, metav1.GetOptions{})
 			if err != nil {
 				return err
 			}
 
+			tempDir, err := ioutil.TempDir("", "stash-cli")
+			if err != nil {
+				return err
+			}
+			defer os.RemoveAll(tempDir)
+
+			// dump secret
+			if err = localDirs.dumpSecret(tempDir, secret); err != nil {
+				return err
+			}
+
 			// configure restic wrapper
 			extraOpt := util.ExtraOptions{
-				SecretDir:   docker.SecretDir,
-				EnableCache: false,
-				ScratchDir:  docker.ScratchDir,
+				SecretDir:   localDirs.secretDir,
 			}
+			// configure setupOption
 			setupOpt, err := util.SetupOptionsForRepository(*repository, extraOpt)
 			if err != nil {
 				return fmt.Errorf("setup option for repository failed")
 			}
 
-			// write secret and config in a temp dir
-			// cleanup whole tempDir dir at the end
-			tempDir, err := ioutil.TempDir("", "stash-cli")
+			// init restic wrapper
+			resticWrapper, err := restic.NewResticWrapper(setupOpt)
 			if err != nil {
 				return err
 			}
-			defer os.RemoveAll(tempDir)
 
-			// prepare local dirs
-			if err = localDirs.prepareSecretDir(tempDir, secret); err != nil {
+			//configure config directory
+			localDirs.configDir = filepath.Join(tempDir, configDirName)
+			//get restic credential from ENV
```

Comment:
// dump restic's environments into `restic-env` file.
// we will pass this env file to restic docker container.